compiler/rustc_codegen_ssa/src/sir.rs

```diff
@@ -392,11 +397,7 @@ impl SirFuncCx<'tcx> {
     where
         T: TypeFoldable<'tcx> + Copy,
     {
-        if let Some(substs) = self.instance.substs_for_mir_body() {
-            self.tcx.subst_and_normalize_erasing_regions(substs, ty::ParamEnv::reveal_all(), value)
-        } else {
-            self.tcx.normalize_erasing_regions(ty::ParamEnv::reveal_all(), *value)
-        }
+        self.instance.subst_mir(self.tcx, value)
```

Comment:
This function was lifted from cranelift and `substs_for_mir_body()` has been removed by upstream.

I looked in the codegen_clif git history to see if @bjorn3 had already handled this, but it doesn't appear so.

What do you reckon @bjorn3? Looks like the right fix? FWIW all of our tests pass.

Reply:
https://github.com/rust-lang/rust/commit/8a8ee1a3ed249e4077a7ce0b88903b493419e7d7 uses `        self.instance.subst_mir_and_normalize_erasing_regions(self.tcx, ty::ParamEnv::reveal_all(), value)`.

(CI on rust-lang/rust ensures that the cg_clif subtree always compiles. Periodically I pull in the changes to the cg_clif subtree into the main repo of cg_clif and vice versa.)

Reply:
Ah, thanks.

I'd been looking in https://github.com/bjorn3/rustc_codegen_cranelift -- the wrong place perhaps?

Reply:
That is the main repo. In https://github.com/rust-lang/rust/pull/77975 it was added as `git subtree` to the rust repo. `git subtree` allows syncing of a directory in a repo with a whole different repo. In this case the `compiler/rustc_codegen_cranelift` directory with the https://github.com/bjorn3/rustc_codegen_cranelift repo.

Reply:
Thanks for clarification.